oppijanumerorekisteri-service/src/main/java/fi/vm/sade/oppijanumerorekisteri/services/impl/HenkiloServiceImpl.java

```diff
@@ -69,53 +86,147 @@ public boolean getOidExists(String oid) {
     @Override
     @Transactional(readOnly = true)
     public String getOidByHetu(String hetu) {
-        return this.henkiloHibernateRepository.findOidByHetu(hetu).orElseThrow(NotFoundException::new);
+        return this.henkiloJpaRepository.findOidByHetu(hetu).orElseThrow(NotFoundException::new);
     }
 
     @Override
     @Transactional(readOnly = true)
     public List<HenkiloPerustietoDto> getHenkiloPerustietoByOids(List<String> oids) {
         List<Henkilo> henkilos = this.henkiloDataRepository.findByOidhenkiloIsIn(oids);
-        return mapper.mapAsList(henkilos, HenkiloPerustietoDto.class);
+        return this.mapper.mapAsList(henkilos, HenkiloPerustietoDto.class);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<HenkiloDto> getHenkilosByOids(List<String> oids) {
+        List<Henkilo> henkilos = this.henkiloDataRepository.findByOidhenkiloIsIn(oids);
+        if(henkilos.isEmpty()) {
+            throw new NotFoundException();
```

Comment:
Pitäisikö tämä tehdä vasta controllerissa? Mielestäni ei ole loogista, että service käyttäytyisi näin.